lib/Conversion/TorchToTosa/TorchToTosa.cpp

```diff
@@ -526,6 +526,41 @@ LogicalResult ConvertAtenOp<AtenArgmaxOp>::matchAndRewrite(
   return success();
 }
 
+template <>
+LogicalResult ConvertAtenOp<AtenSqueezeOp>::matchAndRewrite(
+    AtenSqueezeOp op, OpAdaptor adaptor,
+    ConversionPatternRewriter &rewriter) const {
+
+  Value self = adaptor.self();
+  auto selfTy = self.getType().template cast<RankedTensorType>();
+
+  if (!selfTy)
+    return op.emitError("Only ranked tensor types supported in TOSA argmax");
+
+  auto selfShape = selfTy.getShape();
+
+  SmallVector<int64_t> newOutputShape;
+  for (auto &dim : selfShape) {
```

Comment:
Maybe add a comment here about the assumptions w.r.t. to dimensions dynamically equal to 1 where it is not known statically.

Reply:
Will do. Am planning to refactor this to also support squeeze.dim so next version will also contain your suggestion. May have unsqueeze.dim as well at the time.